bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/ghost/Ghost.scala

```diff
@@ -0,0 +1,100 @@
+package edu.gemini.spModel.gemini.ghost
+
+import java.beans.PropertyDescriptor
+import java.util.{Collections, List => JList, Map => JMap, Set => JSet}
+
+import edu.gemini.pot.sp.{ISPObservation, SPComponentType}
+import edu.gemini.shared.util.immutable.ImList
+import edu.gemini.shared.util.immutable.ScalaConverters._
+import edu.gemini.spModel.core.Site
+import edu.gemini.spModel.data.ISPDataObject
+import edu.gemini.spModel.data.config.{DefaultParameter, DefaultSysConfig, ISysConfig, StringParameter}
+import edu.gemini.spModel.data.property.{PropertyProvider, PropertySupport}
+import edu.gemini.spModel.obscomp.{InstConfigInfo, InstConstants, SPInstObsComp}
+import edu.gemini.spModel.pio.{ParamSet, PioFactory}
+import edu.gemini.spModel.seqcomp.SeqConfigNames
+import edu.gemini.spModel.target.env.AsterismType
+
+import scala.collection.immutable.TreeMap
+import scala.collection.JavaConverters._
+
+
+/** The GHOST instrument SP model.
+  * Note that we do not override clone since private variables are immutable.
+  */
+class Ghost extends SPInstObsComp(Ghost.SP_TYPE) with PropertyProvider {
+  override def getSite: JSet[Site] = {
+    Site.SET_GS
+  }
+
+  override def getPhaseIResourceName: String = {
+    "gemGHOST"
+  }
+
+  override def getProperties: JMap[String, PropertyDescriptor] = {
+    Ghost.PropertyMap
+  }
+
+  /** ParamSet methods.
+    *
+    * We don't have to override these, as SPInstObsComp handles our only property
+    * right now, i.e. POS_ANGLE_PROP, but we do so anyway as we will eventually need
+    * to add additional properties.
+    */
+  override def getParamSet(factory: PioFactory): ParamSet = {
+    super.getParamSet(factory)
+  }
+
+  override def setParamSet(paramSet: ParamSet): Unit = {
+    super.setParamSet(paramSet)
+  }
+
+  override def getSysConfig: ISysConfig = {
+    val sc = new DefaultSysConfig(SeqConfigNames.INSTRUMENT_CONFIG_NAME)
+    sc.putParameter(StringParameter.getInstance(ISPDataObject.VERSION_PROP, getVersion))
+    sc.putParameter(DefaultParameter.getInstance(Ghost.POS_ANGLE_PROP, getPosAngle))
+    sc
+  }
+
+  override def getSupportedAsterismTypes: ImList[AsterismType] = {
+    List(AsterismType.GhostStandardResolution, AsterismType.GhostHighResolution).asImList
+  }
+
+  override def getDefaultAsterismType: AsterismType = {
+    AsterismType.GhostStandardResolution
+  }
+
+  // TODO: Assume we will need to redefine these methods when we have values for them.
+  override def getScienceArea: Array[Double] = ???
+  override def getSetupTime(obs: ISPObservation): Double = ???
```

Comment:
It seems like we can leave this one at the default, 20 mins, for now.  They won't be looking at planned time calculations in June and it'll have to be defined or the OT will blow up.